src/de.rs

```diff
@@ -243,6 +243,48 @@ impl<'de, R: Read<'de>> Deserializer<R> {
         Error::syntax(reason, position.line, position.column)
     }
 
+    /// Returns the first non-comment byte without consuming it, or `None` if
+    /// EOF is encountered.
+    fn parse_comment(&mut self) -> Result<Option<u8>> {
+        // Cursor is already after the first slash.
+        match tri!(self.peek()) {
+            // Line Comment
+            Some(b'/') => {
+                self.eat_char();
+
+                // And eat the rest of the line.
+                loop {
+                    match tri!(self.peek()) {
+                        // Found the end of the line.
+                        // Pass it back to parse_whitespace.
+                        Some(b'\r') | Some(b'\n') => {
+                            self.eat_char();
+                            return self.parse_whitespace()
+                        },
+                        _ => {
+                            self.eat_char();
+                        }
+                    };
+                }
+            }
+
+            // Block Comment
+            Some(b'*') => {
+                self.eat_char();
+
+                // TODO: eat until */.
```

Comment:
Important TODO

Reply:
YUP! But I'm not 100% sure how to do this. The straightforward version is to look for the `*/` byte sequence at `u8` intervals, but I haven't yet convinced myself that it's impossible to create that same byte sequence at non-character boundaries (e.g. four byte chars containing `00101010 00101111`).

```
*/
\u002A \u002F
00101010 00101111
```

Reply:
Okay, I was right, I do need to be charset aware:

```
"\uAA2A\u2F00"
10101010 00101010 00101111 00000000
```

Reply:
(This also means that the loop looking for EOL in the line comment is wrong.)